docs/vgclog_examples.md

```diff
@@ -706,3 +707,895 @@ Analyzing the structure and elements of this example log output shows that this
+
+- As expected, the final global mark phase increment does not reclaim any free memory.
+- The remembered set metastructure has been rebuilt. The count has reduced and the amount of available memory in the metastructure has increased from 91% to 94%.
+- The number of overflow regions remains unchanged. The final global mark phase increment did not manage to rebuild any overflow regions.
+
+Following the final global mark increment, the global mark cycles completes and the GC ends the STW pause.
+
+```xml
+<cycle-end id="1225" type="global mark phase" contextid="1154" timestamp="2021-02-26T11:17:37.034" />
+<exclusive-end id="1226" timestamp="2021-02-26T11:17:37.034" durationms="170.186" />
+
+```
+
+The operations to create a record of object liveness across the heap, which began with the global *mark* cycle, is completed with a sweep phase. The sweep phase is triggered by the end of the global *mark* cycle to be included in the next partial GC cycle that runs.
+
+#### Sweep phase
+
+While the global *sweep* operation is logically associated with the global *mark* phase, it does not run in the same global *mark* cycle. Instead, the *sweep* operation runs in the same STW increment as the first partial GC cycle that runs after the completion of the global *mark* cycle. This can be seen in the following log excerpt.  After the log records the end of the global mark cycle it records a STW pause followed by a `partial gc` cycle of `id=1229`. The global *sweep* operation that runs after the the global *mark* phase is recorded in the `<gc-op>` element tagged as `id=1229`.
+
+```xml
+<exclusive-start id="1227" timestamp="2021-02-26T11:17:38.804" intervalms="1940.125">
+...
+<cycle-start id="1229" type="partial gc" contextid="0" timestamp="2021-02-26T11:17:38.805" intervalms="3926.202" />
+...
+</gc-start>
+...
+</gc-start>
+<gc-op id="1232" type="sweep" timems="9.472" contextid="1229" timestamp="2021-02-26T11:17:38.815" />
+<gc-op id="1233" type="copy forward" timems="308.258" contextid="1229" timestamp="2021-02-26T11:17:39.124">
+...
+<gc-op id="1234" type="classunload" timems="0.012" contextid="1229" timestamp="2021-02-26T11:17:39.125">
+...
+<gc-end>
+...
+</gc-end>
+<cycle-end id="1237" type="partial gc" contextid="1229" timestamp="2021-02-26T11:17:39.125" />
+<exclusive-end id="1238" timestamp="2021-02-26T11:17:39.125" durationms="320.792" />
+```
+
+A record of object liveness is now complete.
+*
+#### Summary of the  `balanced` global mark GC cycle example
+
+Analyzing the structure and elements of this example log output shows that this example `balanced` global mark GC cycle has the following characteristics:
+
+- The GC cycle begins with an STW pause and is triggered by a taxation allocation threshold that has a value equal to half the size of the eden space. Each global mark phase is also triggered by a taxation threshold equal to the size of half of the eden space.
```

Comment:
as mentioned above, there is an additional criteria to start the cycle (beside just meeting 1/2 eden taxation threshold), what is low total free memory .

Reply:
Replaced with:

_Analyzing the structure and elements of this example log output shows that this example `balanced` global mark GC cycle has the following characteristics:

- If the total free memory is low when the taxation allocation threshold is reached, the GC triggers a global mark cycle. The allocation taxation threshold is set by the previous cycle to trigger a new cycle when the eden space is half full. This threshold value frees up eden space to enable a global mark cycle to interleave with the garbage collection operations of partial GC cycles.
- Each global mark phase increment is triggered by an allocation taxation threshold value that is set to half of the eden space.
- Global mark GC cycle and global mark cycle increments begin with a STW pause.
- The global *mark* cycle does not reclaim memory. The cycle creates an updated record of object liveness by attempting to rebuild overflowed and stable regions. The change in status of the remembered set metastructure can be seen in the logs by inspecting the `<remembered-set>` attributes.
- Partial cycles run in between global mark phase increments.
- The final global mark phase increment includes a class unload. The final increment also triggers a sweep phase to run in the next partial cycle._